xarray/backends/netCDF4_.py

```diff
@@ -209,6 +217,8 @@ def __init__(self, filename, mode='r', format='NETCDF4', group=None,
         self._opener = opener
         self._filename = filename
         self._mode = 'a' if mode == 'w' else mode
+        self._unlimited_dimensions = set()
+        self.encoding = None
```

Comment:
I don't think you need an `encoding` property for datastores. Shouldn't `_unlimited_dimensions` be enough?